service/history/fx.go

```diff
@@ -104,27 +99,9 @@ var Module = fx.Options(
 )
 
 func ServiceProvider(
-	grpcServerOptions []grpc.ServerOption,
-	serviceConfig *configs.Config,
-	visibilityMgr manager.VisibilityManager,
-	handler *Handler,
-	logger log.SnTaggedLogger,
-	grpcListener net.Listener,
-	membershipMonitor membership.Monitor,
-	metricsHandler metrics.Handler,
-	healthServer *health.Server,
+	args ServiceArgs,
 ) *Service {
-	return NewService(
-		grpcServerOptions,
-		serviceConfig,
-		visibilityMgr,
-		handler,
-		logger,
-		grpcListener,
-		membershipMonitor,
-		metricsHandler,
-		healthServer,
-	)
+	return NewService(args)
```

Comment:
just get rid of this and use NewService as a provider?

Reply:
Done